src/main/java/darren/assistant/Parser.java

```diff
@@ -0,0 +1,204 @@
+package darren.assistant;
+
+import darren.assistant.commands.*;
+import darren.assistant.tasks.DeadlinesTask;
+import darren.assistant.tasks.EventsTask;
+import darren.assistant.tasks.Task;
+import darren.assistant.tasks.ToDoTask;
+import darren.assistant.ui.DarrenAssistantException;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parses user input (interactive commands) and stored lines (for persistence)
+ * into domain objects used by the application.
+ *
+ * <p>Responsibilities:</p>
+ * <ul>
+ *   <li>Convert raw user commands into concrete {@link Command} objects.</li>
+ *   <li>Parse date/time strings typed by the user into {@link LocalDateTime}.</li>
+ *   <li>Reconstruct {@link Task} objects from storage lines.</li>
+ * </ul>
+ */
+public class Parser {
+    /**
+     * Parses a single line of user input into a {@link Command}.
+     *
+     * <p>Supported commands (case-insensitive):</p>
+     * <ul>
+     *   <li><code>bye</code></li>
+     *   <li><code>list</code></li>
+     *   <li><code>todo &lt;desc&gt;</code></li>
+     *   <li><code>deadline &lt;desc&gt; /by &lt;when&gt;</code></li>
+     *   <li><code>event &lt;desc&gt; /from &lt;start&gt; /to &lt;end&gt;</code></li>
+     *   <li><code>mark &lt;index&gt;</code>, <code>unmark &lt;index&gt;</code>, <code>delete &lt;index&gt;</code></li>
+     * </ul>
+     *
+     * @param line raw user input
+     * @return a concrete {@link Command} to be executed
+     * @throws DarrenAssistantException if the input is empty, malformed,
+     *                                  or corresponds to an unknown command
+     */
+    public static Command parseTask(String line) throws DarrenAssistantException {
+        String input = line.trim();
+        if (input.isEmpty()) {
+            throw new DarrenAssistantException("Empty command.");
+        }
+
+        String low = input.toLowerCase();
+
+        if (low.equals("bye")) {
```

Comment:
Would it be better to do "startsWith" for these commands too in case the user accidentally adds characters behind the command?
```suggestion
        if (low.startsWith("bye")) {
```